\documentclass{article}
\usepackage{geometry}
\usepackage{nopageno}
\geometry{top=1.0in, bottom=1.0in, left=1.0in, right=1.0in}

\newcommand{\ie}{{\em i.e.,}~}
\newcommand{\eg}{{\em e.g.,}~}

\begin{document}
	
\pagestyle{plain}

\textbf{Describe a significant computer science project you have worked on. If you have worked on a major independent research project (such as research for a graduate program), please describe that work here. Give an overview of the problem, explain how you approached key technical challenges, and describe what you gained from the experience. If the project was team-based, be sure to specify your individual role and contributions.}\\

I am affiliated with the AMP Lab at UC Berkeley.
The lab's mission is to use machine learning (\textbf{A}lgorithms), warehouse-scale computing (\textbf{M}achines), and crowdsourcing (\textbf{P}eople) to tackle Big Data problems.
One of my advisors, David Patterson, has argued that it is important to look outside CS for problems to drive innovation in our field.\footnote{http://www.nytimes.com/2011/12/06/science/david-patterson-enlist-computer-scientists-in-cancer-fight.html} % could change this to a reference
Whole-genome analysis, as opposed to selective analysis\footnote{\eg https://www.23andme.com/}, has emerged as a potentially powerful medical diagnostic tool.
Once available only to the likes of Steve Jobs, whole-genome sequencing has rapidly become more affordable; the wet lab portion of genome analysis will soon fall to \$1000. % citation to steve jobs book/nyt article?
With these developments, the data analysis required to transform the raw output of DNA sequencing machines to a full genome has become the bottleneck\footnote{http://www.nytimes.com/2011/12/01/business/dna-sequencing-caught-in-deluge-of-data.html} and is therefore a compelling target for our lab.

The first step in this analysis pipeline is \textit{short read alignment}, where for each read (\ie around 100 bases -- the letters A, T, C, G -- of DNA), we find the location in the reference genome where it best matches.
The reference genome is a 3-billion base aggregate of what human genomes are like; since only 0.1\% of an individual's genome is unique, each read will only differ from the reference by a few characters.
My colleagues and I developed the Scalable Nucleotide Alignment Program (SNAP), an algorithm that leverages resource improvements and conserves computation to get better speed and accuracy than current solutions.

My main contribution was characterizing the patterns in the genome.
What makes alignment hard is that the genome is not random; rather, it is highly redundant, \eg from proteins that multiple chromosomes encode.
Thus, reads often align to many locations throughout the genome, and it is expensive to find the best match.
If it were simply a matter of exact duplication, we could easily (and do) identify it beforehand.
Then, when we detect that a read matches a duplicate region, we give up early, since an unambiguous alignment is impossible.
However, the more difficult problem is near duplication; \ie there are many substrings in the genome that are very similar to each other (\eg a few edits apart).
We still want to identify these in advance; however, since an unambiguous best alignment may exist, we cannot merely quit early.
Instead, we want to compare against them in aggregate rather than doing the na\"{\i}ve thing of comparing against them all, one at a time.
Therefore, I designed and implemented a distributed algorithm for identifying similar substrings throughout the genome.

Through working on this project, I learned a great deal about genomics and the potential for personalized medicine as well as how computer scientists can help with problems like this.
%I learned about choosing applications to drive research so you don't fall victim to the hammer/nail problem (\ie if you have a hammer, everything looks like a nail).
I have broadened my circle of collaborators to include another student, other professors in the lab, a medical doctor at UCSF, and a researcher at MSR.
This networking not only helps with idea generation but also will help me identify job opportunities.
I was invited to give presentations on this work at UC Santa Cruz, Genentech, and SRI and got feedback from people who are experts in genomics.
We released a tech report on the algorithm a few months ago\footnote{http://arxiv.org/abs/1111.5572}, and we plan to submit a full paper in the next few weeks.
Most of all, it has been exciting to work on a project that could have a big impact and help lots of people.

\end{document}